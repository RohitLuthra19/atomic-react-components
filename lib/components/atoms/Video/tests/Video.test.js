import React from 'react';
import { shallow, render } from 'enzyme';

import { VideoVanilla } from '../index';
import defaultConfig from '../Video.mock';

describe('<Video />', () => {
  let VideoComponent = '';
  beforeEach(() => {
    VideoComponent = shallow(<VideoVanilla {...defaultConfig}>Test</VideoVanilla>);
  });

  it('should have video tag', () => {
    console.log(VideoComponent.find('video'));
    expect(VideoComponent.find('video')).toBeTruthy();
  });

  it('should have source tag in video', () => {
    expect(VideoComponent.find('video').find('source')).toBeTruthy();
  });

  it('should have src attribute in source tag', () => {
    const VideoComponentRender = render(
      <VideoVanilla {...defaultConfig}>
        <source src={defaultConfig.src} type="mp4" />
        <track />
      </VideoVanilla>
    );
    expect(VideoComponentRender.find('source').attr('src')).toEqual(defaultConfig.src);
  });

  it('should be autoplay but need to be mute', () => {
    const VideoComponentRender = render(
      <VideoVanilla {...defaultConfig}>
        <source src={defaultConfig.src} />
        <track />
      </VideoVanilla>
    );
    expect(
      VideoComponentRender.find({ autoplay: true }) && VideoComponentRender.find({ mute: true })
    ).toBeTruthy();
  });

  test('should render correctly', () => {
    expect(VideoComponent).toMatchSnapshot();
  });
});
