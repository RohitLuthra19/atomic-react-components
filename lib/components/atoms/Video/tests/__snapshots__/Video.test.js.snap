// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video /> should render correctly 1`] = `
<video
  autoPlay={false}
  className="videoContainer"
  controls={true}
  controlsList="nodownload"
  height="auto"
  loop={true}
  muted={true}
  poster=""
  width="100%"
>
  <source
    src="http://techslides.com/demos/sample-videos/small.mp4"
    type="video/mp4"
  />
  <Track
    id={1}
    isDefault={true}
    kind="subtitles"
    label="English"
    src="/"
    srcLang="en"
  />
</video>
`;
